hack/golangci-strict.yaml

```diff
@@ -83,6 +93,12 @@ issues:
         - gosimple
       text: "S1033: unnecessary guard around call to delete"
 
+    # Only packages listed here opt into the strict "exported symbols must be documented".
+    - linters:
+        - revive
+      text: "(exported: exported .* or be unexported|exported: comment on exported.*should be of the form)"
+      path-except: cmd/kubeadm
```

Comment:
like i mentioned on slack due to the generated* files in cmd/kubeadm/app/apis we have to skip that folder as well.
it contains some function names with CamelCase and underscores that did not pass this type of lint...or at least golint.

generally  i am +1 for such a change, but if the overall setup requirement is -1 from sig arch/testing, then i guess it can't be enabled.

Reply:
yeah, it's failing:
> cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go:35:1: exported: exported function Convert_v1beta3_InitConfiguration_To_kubeadm_InitConfiguration should have comment or be unexported (revive)

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121504/pull-kubernetes-verify-lint/1717127788919525376

Reply:
or if it's possible to skip a zz_generated* file pattern that would work too..
https://github.com/kubernetes/kubernetes/tree/master/cmd/kubeadm/app/apis/kubeadm
https://github.com/kubernetes/kubernetes/tree/master/cmd/kubeadm/app/apis/kubeadm/v1beta4

Reply:
`cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go` isn't generated. Files with a comment saying "generated" are already skipped by golangci-lint automatically.

Reply:
If we want to go ahead with this, then a separate PR which fixes all of these findings would be good.

Reply:
i can send the cleanup later today, but i wanted to understand if this has green light.

Reply:
here is the PR, it can be merged regardless.
https://github.com/kubernetes/kubernetes/pull/121511

thanks for creating the config in this PR. i used it locally to fix the problems.